bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1101,13 +1102,38 @@ fn gen_transaction_id() -> String {
 /// is dropped, the file will be removed from the disk.
 #[derive(uniffi::Object)]
 pub struct MediaFileHandle {
-    inner: SdkMediaFileHandle,
+    inner: RwLock<Option<SdkMediaFileHandle>>,
```

Comment:
What's this for?
Why do we need it now and we didn't before?

Reply:
Before, the `SdkMediaFileHandle` was owned and never changed during the lifetime of a `MediaFileHandle`. Now the patch changes that: when calling `persist()`, we "steal" the content of the `SdkMediaFileHandle`, because `SdkMediaFileHandle::persist` requires an owned `self` (and not borrowed as in `&self`).

Because of the stealing, two new things are now true that weren't before:

- the handle can be missing, so it must be `Option`al
- since we require write access to steal (as `Option::take` requires ownership of the option), and we are in a multithreaded context (we want the struct to be `Send` and/or `Sync`), that means having a lock.

This is related to my review comments: if the handle is missing after a call to `persist`, it's actually possible to crash the app via misuse of this API: the following sequence would cause a crash: `handle.persist(); handle.path();`, which I would prefer to be an error.